engines/windowsaio.c

```diff
@@ -322,34 +343,67 @@ static int fio_windowsaio_getevents(struct thread_data *td, unsigned int min,
 		end_count = start_count + (t->tv_sec * 1000) + (t->tv_nsec / 1000000);
 	}
 
-	do {
-		io_u_qiter(&td->io_u_all, io_u, i) {
-			if (!(io_u->flags & IO_U_F_FLIGHT))
-				continue;
+	if (o->no_completion_thread) {
```

Comment:
Please split this into two functions, ala:

```
if (o->no_completion_thread)
    return fio_windowsaio_getevents_nothread(...);
return fio_windowaio_getevents_thread();
```

or something like that.

Reply:
Updated per comments.